gattc_windows.go

```diff
@@ -121,3 +121,103 @@ type DeviceService struct {
 func (s *DeviceService) UUID() UUID {
 	return s.uuidWrapper
 }
+
+// DiscoverCharacteristics discovers characteristics in this service. Pass a
+// list of characteristic UUIDs you are interested in to this function. Either a
+// list of all requested characteristics is returned, or if some characteristics could not be
+// discovered an error is returned. If there is no error, the characteristics
+// slice has the same length as the UUID slice with characteristics in the same
+// order in the slice as in the requested UUID list.
+//
+// Passing a nil slice of UUIDs will return a complete
+// list of characteristics.
+func (s *DeviceService) DiscoverCharacteristics(uuids []UUID) ([]DeviceCharacteristic, error) {
+	getCharacteristicsOp, err := s.service.GetCharacteristicsWithCacheModeAsync(bluetooth.BluetoothCacheModeUncached)
+	if err != nil {
+		return nil, err
+	}
+
+	// IAsyncOperation<GattCharacteristicsResult>
+	if err := awaitAsyncOperation(getCharacteristicsOp, genericattributeprofile.SignatureGattCharacteristicsResult); err != nil {
+		return nil, err
+	}
+
+	res, err := getCharacteristicsOp.GetResults()
+	if err != nil {
+		return nil, err
+	}
+
+	gattCharResult := (*genericattributeprofile.GattCharacteristicsResult)(res)
+
+	// IVectorView<GattCharacteristic>
+	charVector, err := gattCharResult.GetCharacteristics()
+	if err != nil {
+		return nil, err
+	}
+
+	// Convert characteristics vector to array
+	servicesSize, err := charVector.GetSize()
+	if err != nil {
+		return nil, err
+	}
+
+	var characteristics []DeviceCharacteristic
+	for i := uint32(0); i < servicesSize; i++ {
+		s, err := charVector.GetAt(i)
+		if err != nil {
+			return nil, err
+		}
+
+		characteristic := (*genericattributeprofile.GattCharacteristic)(s)
+		guid, err := characteristic.GetUuid()
+		if err != nil {
+			return nil, err
+		}
+
+		serviceUuid := winRTUuidToUuid(guid)
+
+		// add if in our original list
+		addToList := len(uuids) == 0
+
+		for _, uuid := range uuids {
+			if serviceUuid.String() == uuid.String() {
+				// one of the services we're looking for.
+				addToList = true
+				break
+			}
+		}
+
+		properties, err := characteristic.GetCharacteristicProperties()
```

Comment:
Why not just put all this logic inside the loop instead of having the `addToList` variable?

Reply:
Because we also have to include them if the UUID filter is nil. But now that I think about it the way this is done on Linux and Darwin is much simple. So let me change it.